src/fs.rs

```diff
@@ -604,6 +608,15 @@ impl<IO: Read + Write + Seek, TP, OCC> FileSystem<IO, TP, OCC> {
         Ok(())
     }
 
+    pub fn set_fat_dirty_flag(&self, dirty: bool) -> Result<(), Error<IO::Error>> {
```

Comment:
I think handling it when filesystem is mounted would be better. Fail by default if dirty flag is set + add option in `FsOptions` to allow mounting a dirty FS (and clear the flag during mounting - this probably deserves another option). Also handle dirty flag from BSP. I don't like showing in user-facing API those weird and legacy things from FAT like doubled dirty flags by making two setters and getters.
Of course it is still a bad to work with a dirty FS. Proper solution would be to add some filesystem checking functionality but that's probably a ton of work...